### PR TITLE
bugfix:select for update exception

### PR DESCRIPTION
**Pull request by philoli11**
### Ⅰ. Describe what this PR did
fix select for update exception

### Ⅱ. Does this pull request fix one issue?

#920

### CODE REVIEW

File: core/src/main/java/io/seata/core/protocol/MergeResultMessage.java [hunk 1]
```diff
@@ -116,7 +116,11 @@ public void decode(ByteBuffer byteBuffer) {
     @Override
     public String toString() {
         StringBuilder sb = new StringBuilder("MergeResultMessage ");
-        for (AbstractMessage msg : msgs) { sb.append(msg.toString()).append("\n"); }
+        if (msgs != null) {
```
Comment (CoffeeLatte007) [inline core/src/main/java/io/seata/core/protocol/MergeResultMessage.java]: if(msg == null){
return sb.toString
}
for (AbstractMessage msg : msgs) {
                sb.append(msg.toString()).append("\n");
            }
       
        return sb.toString();

Comment (leizhiyuan) [inline core/src/main/java/io/seata/core/protocol/MergeResultMessage.java]: which sence? the msg is null ,and we need to return MergeResultMessage to client? maybe there are some bugs ?

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/SelectForUpdateExecutor.java [hunk 2]
```diff
@@ -106,6 +106,9 @@ public T doExecute(Object... args) throws Throwable {
 
                     TableRecords selectPKRows = TableRecords.buildRecords(getTableMeta(), rsPK);
                     String lockKeys = buildLockKey(selectPKRows);
+                    if (lockKeys == null) {
```
Comment (CoffeeLatte007) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/SelectForUpdateExecutor.java]: StringUtils

Comment (philoli11) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/SelectForUpdateExecutor.java]: done

File: core/src/main/java/io/seata/core/protocol/MergeResultMessage.java (hunk 1 above)
Comment (philoli11) [reply core/src/main/java/io/seata/core/protocol/MergeResultMessage.java]: I simulated the transfer sence, NPE will appear when testing with multiple threads.

**Pull request merged**